### PR TITLE
configuration: Add CPU tab to game properties and slight per-game settings rework

### PR DESCRIPTION
**Pull request by lat9nq**
Originally I had told myself not to add this bit to per-game settings, as the CPU tab was a place for debug settings. As time went on, settings were add to this tab that apply fixes to specific games, thus the need for this.

This change allows setting CPU accuracy to Accurate or Unsafe per-game, as well as the accuracy options for Unsafe. Debug is not allowed here as a per-game CPU accuracy.

Also took this opportunity to lightly rework the other per-game settings. Code should be a tad easier to maintain in those regards.

This needs testing.

### CODE REVIEW

**Comment by lat9nq** [comment]
This now also demotes `custom_rtc` from per-game settings. It's currently broken, and according to @Morph1984 a replacement of sorts should be on its way.

File: src/yuzu/configuration/configuration_shared.cpp [hunk 1]
```diff
@@ -13,32 +13,44 @@
...
+        if (combobox->currentIndex() == ConfigurationShared::USE_GLOBAL_INDEX) {
+            setting->SetGlobal(true);
+        } else {
+            setting->SetGlobal(false);
+            setting->SetValue(combobox->currentIndex() - ConfigurationShared::USE_GLOBAL_OFFSET);
+        }
     }
 }
 
 void ConfigurationShared::ApplyPerGameSetting(Settings::Setting<Settings::RendererBackend>* setting,
```
Comment (ameerj) [inline src/yuzu/configuration/configuration_shared.cpp:42]: You can probably use a templated function for the `ApplyPerGameSetting` and reduce the code duplication.

Comment (lat9nq) [reply src/yuzu/configuration/configuration_shared.cpp:42]: I need to axe the unused version of `ApplyPerGameSetting`, but the remaining versions are not sharing exactly the same code.

File: src/yuzu/configuration/configure_cpu.cpp [hunk 2]
```diff
@@ -54,16 +67,42 @@ void ConfigureCpu::AccuracyUpdated(int index) {
...
-    ui->unsafe_group->setVisible(static_cast<Settings::CPUAccuracy>(index) ==
-                                 Settings::CPUAccuracy::Unsafe);
+    if (Settings::IsConfiguringGlobal()) {
+        ui->unsafe_group->setVisible(static_cast<Settings::CPUAccuracy>(index) ==
+                                     Settings::CPUAccuracy::Unsafe);
+    } else {
+        ui->unsafe_group->setVisible(
+            static_cast<Settings::CPUAccuracy>(index - ConfigurationShared::USE_GLOBAL_OFFSET) ==
+            Settings::CPUAccuracy::Unsafe);
+    }
```
Comment (ameerj) [inline src/yuzu/configuration/configure_cpu.cpp:77]:
```suggestion
    if (!Settings::IsConfiguringGlobal()) {
        index -= ConfigurationShared::USE_GLOBAL_OFFSET;
    }
    const auto setting = static_cast<Settings::CPUAccuracy>(index);
    ui->unsafe_group->setVisible(setting == Settings::CPUAccuracy::Unsafe);
```

**Review by ameerj — COMMENTED** [review]
Just one last nit

**Review by ameerj — APPROVED** [review]

**Review by bunnei — APPROVED** [review]

**Pull request merged**